src/phase2.hpp

```diff
@@ -163,61 +167,49 @@ Phase2Results RunPhase2(
         {
             uint8_t const* entry = disk.Read(read_cursor, entry_size);
 
-            uint64_t entry_pos = 0;
-            uint64_t entry_offset = 0;
             uint64_t entry_f7 = 0;
+            uint64_t entry_pos_offset;
             if (table_index == 7) {
                 // table 7 is special, we never drop anything, so just build
                 // next_bitfield
                 entry_f7 = Util::SliceInt64FromBytes(entry, 0, k);
-                entry_pos = Util::SliceInt64FromBytes(entry, k, pos_size);
-                entry_offset = Util::SliceInt64FromBytes(entry, k + pos_size, kOffsetSize);
+                entry_pos_offset = Util::SliceInt64FromBytes(entry, k, pos_offset_size);
             } else {
                 // skipping
                 if (!current_bitfield.get(read_index)) continue;
 
-                entry_pos = Util::SliceInt64FromBytes(entry, 0, pos_size);
-                entry_offset = Util::SliceInt64FromBytes(entry, pos_size, kOffsetSize);
+                entry_pos_offset = Util::SliceInt64FromBytes(entry, 0, pos_offset_size);
             }
 
+            uint64_t entry_pos = entry_pos_offset >> kOffsetSize;
+            uint64_t entry_offset = entry_pos_offset & ((1U << kOffsetSize) - 1);
+
             // assemble the new entry and write it to the sort manager
 
             // map the pos and offset to the new, compacted, positions and
             // offsets
             std::tie(entry_pos, entry_offset) = index.lookup(entry_pos, entry_offset);
+            entry_pos_offset = (entry_pos << kOffsetSize) | entry_offset;
 
+            uint8_t bytes[16];
             if (table_index == 7) {
                 // table 7 is already sorted by pos, so we just rewrite the
                 // pos and offset in-place
+                uint128_t new_entry = (uint128_t)entry_f7 << f7_shift;
+                new_entry |= (uint128_t)entry_pos_offset << t7_pos_offset_shift;
+                Util::IntTo16Bytes(bytes, new_entry);
```

Comment:
is there a good reason to not do this byteswap in-place?
Also, if we always write a full 16 byte entries here, is there a good reason not to leave it in the host-endian?

Reply:
We don't always write 16 bytes here, the entry size is determined by `EntrySizes::GetMaxEntrySize`, which returns `3*k+1` bits rounded up to whole bytes. There's also a convention in the codebase that all integers are encoded as big-endian before being written to disk, including in `SortManager` which assumes this as well for sorting entries.